### DISCUSSION TITLE
CHANNEL ERRORS - What are they?

### DISCUSSION THREAD

**Discussion by kerbo** [supabase/supabase#22153] [Questions]
I often get Channel Errors. There is no information on what they are and under what conditions you would get them.

In my case, I am using the Flutter API, Realtime database channels, running using the Ios simulator and connecting to the Supabase server (not running locally). The channel code looks like:

channel.onPostgresChanges (.....).subscribe(status, error) {...}

Whenever the status is equal to RealtimeSubscribeStatus.channelError, I log the event to my console. The error value is always NULL. Errors usually occur when the channel is open and the application is idle (like overnight).

My questions are:

1. What are Channel Errors?
2. Under what conditions would Supabase throw a Channel error?
3. How can you get more information on the actual error?
4. Is it ok to ignore these events?

**Comment by GaryAustin1** [comment]
Realtime can get errors for many reasons. Tabs going into background, network lost, power saving modes of mobile devices.  Realtime will sometimes retry and reconnect.  But if an event occurs during the reconnect process or while realtime is not connected it will be lost as there is no queue.  

This repository and older discussion look at this. 
https://github.com/GaryAustin1/Realtime2
https://github.com/orgs/supabase/discussions/5641

**Reply by kerbo** [reply]
I concluded from looking at your links that if you get an error just reconnect the channel...that is if the status is NOT subscribed then unsubscribe (removeChannel) and subscribe again. The actual reason for the error is unimportant.

**Reply by GaryAustin1** [reply]
Yes. I don't try and figure out why. But I don't just reconnect as changes are missed. I refetch any missed data.  I also just shut things down when errors occur and visibility is not set as waste of a connection to keep trying.
